pkg/controller/jaeger/dependencies.go

```diff
@@ -44,19 +45,18 @@ func (r *ReconcileJaeger) handleDependencies(ctx context.Context, str strategy.S
 }
 
 func (r *ReconcileJaeger) handleDependency(ctx context.Context, str strategy.S, dep batchv1.Job) error {
-	tracer := global.TraceProvider().GetTracer(v1.ReconciliationTracer)
+	tracer := otel.GetTracerProvider().Tracer(v1.ReconciliationTracer)
 	ctx, span := tracer.Start(ctx, "handleDependency")
 	defer span.End()
 
 	span.SetAttributes(
-		key.String("dependency.name", dep.Name),
-		key.String("dependency.namespace", dep.Namespace),
+		otelattribute.String("dependency.name", dep.Name),
+		otelattribute.String("dependency.namespace", dep.Namespace),
 	)
 
 	err := r.client.Create(ctx, &dep)
 	if err != nil && !apierrors.IsAlreadyExists(err) {
-		span.SetStatus(codes.Internal)
-		span.SetAttribute(key.String("error", err.Error()))
+		span.SetStatus(codes.Error, err.Error())
```

Comment:
Looks like `tracing.HandleError` is already taking care of the `span.SetStatus`? You can probably just remove this line (and the same for the other places).